governance/program/src/state/proposal.rs

```diff
@@ -259,6 +259,22 @@ impl ProposalV2 {
         Ok(())
     }
 
+    /// Checks the Proposal was finalized (no more state transition will happen)
+    pub fn assert_is_final_state(&self) -> Result<(), ProgramError> {
+        match self.state {
+            ProposalState::Completed
+            | ProposalState::Cancelled
+            | ProposalState::Defeated
+            | ProposalState::Vetoed => Ok(()),
+            ProposalState::Executing
+            | ProposalState::ExecutingWithErrors
+            | ProposalState::SigningOff
+            | ProposalState::Voting
+            | ProposalState::Draft
+            | ProposalState::Succeeded => Err(GovernanceError::InvalidStateNotFinal.into()),
```

Comment:
nit: Please add a comment here that in the current version due to the state transition bug non executable proposals will be stuck in `Succeeded` state